management/users/users.js

```diff
@@ -125,66 +136,181 @@ WDN.initializePlugin('modal', [function() {
 						method: 'POST',
 						url: '/server/management/users/users.php',
 						data: {
-							callback: 'updateUserAbilities',
+							action: 'updateUserAbilities',
 							userId: userId,
 							removeAbilityArr: removeAbilityArr,
 							addAbilityArr: addAbilityArr
 						},
 						success: function (response) {
-							if (response.success) {
-								refreshUserTable();
-							} else {
-								refreshUserTable();
-								alert(response.error);
+							if (!response || !response.success) {
+								alert(response.error || 'There was an error updating the abilities. Please refresh the page and try again');
 							}
+							refreshUserTable();
 						}
 					});
 				});
-			});
+			};
+
+			function initializeEditButtonEventHandlers() {
+				// Initialize the modal handlers
+				$('.close-modal-button').click(function() {
+					$.colorbox.close();
+				});
+
+				// Initialize the edit button handlers
+				$('#user-management-table').on('click', '.userEditButton', function(event) {
+					event.preventDefault();
+
+					// Initialize the colorbox
+					$('#user-management-table .userEditButton').colorbox({
+						inline: true,
+						width: '50%'
+					});
+
+					// Get user information
+					const userId = $(this).parents('tr').data('user-id');
+					getUserInformation(userId, function(userData) {
+						$('#editFirstName').val(userData.firstName);
+						$('#editLastName').val(userData.lastName);
+						$('#editEmail').val(userData.email);
+						$('#editPhoneNumber').val(userData.phoneNumber);
+					});
+					
+					// Submit user information
+					$('#edit-user-form').on('submit', function(event) {
+						event.preventDefault();
+						$('#edit-user-form button[type=submit]').prop('disabled', true);
+
+						let isValid = true;
+						$(this).find('input').each(function() {
+							const noValue = !$(this).val();
+							if (noValue) {
+								isValid = false;
+							}
+							$(this).toggleClass('is-invalid', noValue);
+						});
 		
-			function refreshUserTable(){
+						const isValidEmail = $('#editEmail').val().match(/.+@.+\..+/);
+						if (isValidEmail == null) {
+							isValid = false;
+							$('#editEmail').addClass('is-invalid');
+						}
+
+						if (isValid) {
+							const newFirstName = $('#editFirstName').val();
+							const newLastName = $('#editLastName').val();
+							const newEmail = $('#editEmail').val();
+							const newPhoneNumber = $('#editPhoneNumber').val();
+
+							updateUserInformation(userId, newFirstName, newLastName, newEmail, newPhoneNumber, function() {
+								$.colorbox.close();
+								$('#edit-user-form button[type=submit]').prop('disabled', false);
+							});
+
+							// Need to remove the event handlers so that multiple don't exist after submitting the form
+							$(this).off();
+						} else {
+							$('#edit-user-form button[type=submit]').prop('disabled', false);
+						}
+					});
+				});
+			};
+
+			function getUserInformation(userId, callbackFunction) {
+				$.ajax({
```

Comment:
Instead of passing around callbacks, a cleaner solution would be javascript promises.
You can essentially just return the ajax call, then do `.then` to get the response.

```
return $.ajax({
		dataType: 'json',
		method: 'GET',
		url: '/server/management/users/users.php',
		data: {
			action: 'getUserInformation',
		userId: userId
	})
       .then((response) => {
           if (!response || !response.success) {
                 alert('whatever')
                 return
           }
           return response.user;
       })
```
then back when you call it..
```
getUserInformation(userId)
      .then((userData) => {
           $('#editFirstName').val(userData.firstName);
           // ... the rest of them
      })
```